assets/src/dashboard/app/router/routerProvider.js

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+
+import { createContext, useRef, useMemo, useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import queryString from 'query-string';
+import { createHashHistory } from 'history';
+
+export const RouterContext = createContext({ state: {}, actions: {} });
+
+function RouterProvider({ children, ...props }) {
+  const history = useRef(props.history || createHashHistory());
```

Comment:
As a follow up for my info: why are we using the "hash" history and not the push-state?